tests/transformations/test_ChannelsLast.py

```diff
@@ -0,0 +1,214 @@
+
+
+def download_model(test_model):
+    if test_model == "FINN-CNV_W2A2":
+        qonnx_url = "https://drive.google.com/uc?id=1BAyYM3N28fDHSWMgK8dZ3kVUsxGZLO3U&export=download"
+
+    elif "RadioML_VGG10":
+        qonnx_url = (
+            "https://github.com/Xilinx/brevitas-radioml-challenge-21/raw/"
+            "9eef6a2417d6a0c078bfcc3a4dc95033739c5550/sandbox/notebooks/models/pretrained_VGG10_w8a8_20_export.onnx"
+        )
+    else:
+        raise ValueError(f"Model called {test_model} is not supported")
+
+    # download test data
+    dl_dir = "/tmp"
+    dl_file = dl_dir + f"/{test_model}.onnx"
+    urllib.request.urlretrieve(qonnx_url, dl_file)
+    # run cleanup with default settings
+    out_file = dl_dir + f"/{test_model}_clean.onnx"
+    cleanup(dl_file, out_file=out_file)
+    return out_file
+
+
+def get_golden_in_and_output(onnx_file, test_model):
+    if test_model == "FINN-CNV_W2A2":
+        input_shape = (1, 3, 32, 32)
+    elif "RadioML_VGG10":
+        input_shape = (1, 2, 1024)
+    else:
+        raise ValueError(f"Model called {test_model} is not supported")
+    input_tensor = np.ones(np.prod(np.asarray(input_shape)), dtype=np.float32)
+    input_tensor = input_tensor.reshape(input_shape)
+
+    model = ModelWrapper(onnx_file)
+    input_dict = {model.graph.input[0].name: input_tensor}
+    golden_output_dict = oxe.execute_onnx(model, input_dict, True)
+    golden_result = golden_output_dict[model.graph.output[0].name]
+
+    return input_tensor, golden_result
+
+
+def analysis_testing_for_chanLast_domain(model):
+    # Define for each ChannelsLast operation the number of minimum dimensions, it needs to have
+    ChanLast_node_types_and_min_dim_input = {
+        "Conv": 3,
+        "MaxPool": 3,
+        "BatchNormalization": 3,
+    }
+    # Check that all wrapped_ops in the registry have a definition here
+    chanLast_op_types = list(ChannelsLast.custom_op.keys())
+    testable_op_types = list(ChanLast_node_types_and_min_dim_input.keys())
+    for op_name in chanLast_op_types:
+        assert (
+            op_name in testable_op_types
+        ), f"The channelsLast op {op_name} is missing a definition for the domain string test."
+
+    for n_type, min_dim in ChanLast_node_types_and_min_dim_input.items():
+        nodes = model.get_nodes_by_op_type(n_type)
+        for n in nodes:
+            input_shape = model.get_tensor_shape(n.input[0])
+            if len(input_shape) >= min_dim:
+                assert (
+                    n.domain == "qonnx.custom_op.ChannelsLast"
+                ), f"Node domain is not set correctly for node with name: {n.name}"
+    return dict()
+
+
+def verify_all_nodes(model):
+    result = dict()
+    for n in model.graph.node:
+        if is_finn_op(n.domain):
+            n_instance = getCustomOp(n)
+            verify_result = n_instance.verify_node()
+            result[n.name] = verify_result
+    return result
+
+
+def analysis_first_node_is_transpose(model):
```

Comment:
consider adding an option to also call `MakeInputChannelsLast` as part of the channels-last transformation: https://github.com/Xilinx/finn-base/blob/dev/src/finn/transformation/make_input_chanlast.py#L35

Reply:
note that in this case the global input tensor shape itself will change and there should be no more transpose node at the beginning once everything is collapsed

Reply:
Yeah, it now adds two test cases, but I think that's fine.